spec/rspec/schema_merger/schema_cleaner_spec.rb

```diff
@@ -0,0 +1,123 @@
+require 'spec_helper'
+require 'json'
+require 'rspec/openapi/schema_cleaner'
+
+RSpec.describe 'SchemaCleaner' do
```

Comment:
Can you test it by only adding YAML-level tests at `spec/rspec/rails_spec.rb`? The only thing that must be maintained in this repository is what OpenAPI schema is generated at each rspec execution, not internal methods to implement it. If you unit-test internal implementations, it becomes harder to refactor the implementation. 

So I'm not comfortable having this kind of test when you could just write a test that fails only when needed. This test could fail when it doesn't need to, for example when we rename `paths_to_all_fields` to something else, which users don't care about. In fact, I actually don't appreciate tests under `spec/rspec/schema_merger/` because the interface of `RSpec::OpenAPI::SchemaMerger.merge!` isn't part of the contact of this gem. Testing the same thing by reading a YAML file generated by running rspec would be more cost-effective because it would fail only when the actual contract is changed.

Reply:
I understand your concerns and reasoning.
However, it is unclear for me to write tests that verify item are removed (or preserved) as expected.
IIUC, rspec-openapi read the spec file if it exist, then updates it.
To check item removal, it is required to read the old spec file with outdated items, but the file itself is updated after rails-rspec, so you or future contributors get confused about how to deal with changes in the file.
Any guidance is appreciated

Reply:
A simple idea occurred to me.
I will erase unit tests and write rails-rspec-based tests

Reply:
I'm okay with having a YAML fixture, like what you did at `spec/rspec/schema_merger/*.json`, for comparing it against an intermediate result of the schema file. If you have two YAML files for the same spec, initializing the destination path with one of the files, running rspec, and comparing the result with the other file wouldn't be too difficult.